### PR TITLE
Standardise TypeScript imports using parser AST

### PR DESCRIPTION
**Pull request by yazshel**
This change set improves handling of TypeScript imports and resolves a few issues:
1. Correctly parses aliased property imports in TypeScript (ie `import {func1 as func1Alias} from 'module1';`)
2. Sort imported properties in a defined way to improve caching (in both TS and JS parsers)
3. Now disregards multiple whitespace in TS import statements to improve caching.

There are several commits covering the various issues; let me know if you'd like me to split, rebase or consolidate the branch in any way.

Timshel

### CODE REVIEW

**Comment by yazshel** [comment]
The failed tests above were failing locally for me on `master` **before** I made any changes, so I'm wondering if an updated package or other external change may have triggered these failures? I had a quick look but wasn't able to identify why the import sizes no longer match what is expected.

Can you trigger a re-test on `master` to verify please?

Cheers,

Timshel

**Comment by yazshel** [comment]
I've added some `console.log` locally to try to debug the test failures, and it appears that  there is a fair size overhead when `webpack` compiles the test script to a `bundle.js`:

image

This example is of the `import-shaken.js` - and the import has clearly been shaken during compilation but the `webpack` overhead pushes the size over the expected 1000 bytes.

Has there been a change to `webpack` that may have triggered this?

**Comment by yairhaimo** [comment]
Thanks a lot for sending a PR! I will go over it as soon as possible.
The latest change in regards to webpack was to upgrade it to version 4, that might be the breaking change. I'll check it out.
Thanks again.

File: packages/import-cost/src/jsParser.js [hunk 1]
```diff
@@ -57,7 +57,16 @@ function parse(source) {
 function compileImportString(node) {
   let importSpecifiers, importString;
   if (node.specifiers && node.specifiers.length > 0) {
-    importString = node.specifiers
+    importString = [].concat(node.specifiers)
+      .sort((s1, s2) => {
+        if (s1.type !== s2.type) {
+          return s1.type < s2.type ? -1 : 1; // ImportDefaultSpecifier comes before ImportNamespaceSpecifier & ImportSpecifiers...
```
[the remark this answers is not in the document]
Comment (yazshel) [reply packages/import-cost/src/jsParser.js]: Thanks, I've just pushed an updated branch with some rebased commits containing several improvements to the original PR.

I've greatly simplified the sort function for JavaScript imports - on thinking it through, for mixed imports, only the NamedImports section needs to be sorted, so the only time we need to sort elements in relation to each other is when **both** items being compared are named imports / ImportSpecifiers. I've added a large comment with an explanation.

I've also improved the TypeScript import normalisation - it did not correctly handle the mixed "DefaultImport, * as NamespaceImport" case in its original form.

Please review and let me know if there is anything else you would like me to change or reconsider.

**Comment by yairhaimo** [comment]
Looks great. Thanks! 🏆

**Pull request merged**